test/cpp/end2end/xds/xds_end2end_test_lib.cc

```diff
@@ -81,11 +81,15 @@ void XdsEnd2endTest::ServerThread::XdsServingStatusNotifier::
   std::map<std::string, grpc::Status>::iterator it;
   while ((it = status_map.find(uri)) == status_map.end() ||
          it->second.error_code() != expected_status) {
-    ASSERT_FALSE(cond_.WaitWithDeadline(&mu_, deadline))
-        << "\nExpected status: " << expected_status << "\nActual:"
-        << (it == status_map.end() ? "Entry not found in map"
-                                   : absl::StrCat(it->second.error_code()));
+    if (cond_.WaitWithDeadline(&mu_, deadline)) {
+      LOG(ERROR) << "\nExpected status: " << expected_status << "\nActual:"
```

Comment:
This message should probably say something like "timeout elapsed" to make it clear that the expected event didn't happen on time.